### PR TITLE
Internet skipper, test fix, and tidy

### PR DESCRIPTION
**Pull request by bjlittle**

### CODE REVIEW

**Comment by bjlittle** [comment]
Ping @marqh and @dkillick

File: lib/iris/tests/__init__.py [hunk 1]
```diff
@@ -275,8 +281,7 @@ def result_path(self, basename=None, ext=''):
                               filename)
         return result
 
-    def assertCMLApproxData(self, cubes, reference_filename=None, *args,
-                            **kwargs):
+    def assertCMLApproxData(self, cubes, reference_filename=None, **kwargs):
```
Comment (bjlittle) [inline lib/iris/tests/__init__.py]: `*args` is not used here

File: lib/iris/tests/__init__.py [hunk 2]
```diff
@@ -289,7 +294,7 @@ def assertCMLApproxData(self, cubes, reference_filename=None, *args,
             if fname[-1].endswith(".cml"):
                 fname[-1] = fname[-1][:-4]
             fname[-1] += '.data.%d.json' % i
-            self.assertCubeDataAlmostEqual(cube, fname, *args, **kwargs)
+            self.assertCubeDataAlmostEqual(cube, fname, **kwargs)
```
Comment (bjlittle) [inline lib/iris/tests/__init__.py]: `*args` is not used here

File: lib/iris/tests/__init__.py [hunk 3]
```diff
@@ -412,13 +416,14 @@ def assertCubeDataAlmostEqual(self, cube, reference_filename, *args,
...
-                                     math.isnan(cube.data.mean()))
+                                   stats.get('max', 0.), stats.get('min', 0.)),
+                                  dtype=np.float_)
+                if math.isnan(stats.get('mean', 0.)):
+                    self.assertTrue(math.isnan(cube.data.mean()))
                 else:
                     cube_stats = np.array((cube.data.mean(), cube.data.std(),
-                                           cube.data.max(), cube.data.min()))
+                                           cube.data.max(), cube.data.min()),
+                                          dtype=np.float_)
```
Comment (bjlittle) [inline lib/iris/tests/__init__.py]: A simple tidy to align what `dtype` of what is saved and what is compared i.e. `np.float_`

File: lib/iris/tests/__init__.py [hunk 4]
```diff
@@ -1042,6 +1047,11 @@ class MyPlotTests(test.GraphicsTest):
     'Test(s) require "nc_time_axis", which is not available.')
 
 
+skip_inet = unittest.skipIf(not INET_AVAILABLE,
+                            ('Test(s) require an "internet connection", '
+                             'which is not available.'))
+
+
```
Comment (bjlittle) [inline lib/iris/tests/__init__.py]: Define the skipper ...

File: lib/iris/tests/test_image_json.py [hunk 5]
```diff
@@ -70,6 +70,7 @@ def run(self):
             self.queue.task_done()
 
 
+@tests.skip_inet
```
Comment (bjlittle) [inline lib/iris/tests/test_image_json.py]: Use the skipper!

Tested on a non-connected host

File: lib/iris/tests/test_basic_maths.py [hunk 6]
```diff
@@ -632,7 +634,7 @@ def test_log2(self):
 
     def test_log10(self):
         e = iris.analysis.maths.log10(self.cube)
-        self.assertCMLApproxData(e, ('analysis', 'log10.cml'))
+        self.assertCMLApproxData(e, ('analysis', 'log10.cml'), rtol=1e-6))
```
Comment (bjlittle) [inline lib/iris/tests/test_basic_maths.py]: Need this for the `Cray` ... don't think it's worth digging to discover - maybe platform or `numpy` build related.

It's a singlton glitch ...

Comment (bjlittle) [inline lib/iris/tests/test_basic_maths.py]: Cut'n paste trauma ... guilty.

**Comment by bjlittle** [comment]
Proof the skipper is also behaving for a host that is connected ... see the travis output for [TestImageFile](https://travis-ci.org/SciTools/iris/jobs/178864280#L4619) unit test.

**Comment by marqh** [comment]
this looks fine and dandy to me

thanks @bjlittle

**Pull request merged**